features/comments_and_kudos/add_comment.feature

```diff
@@ -117,6 +117,22 @@ Scenario: Comment threading, comment editing
     And I should not see "This should be nested" within ".thread .thread .thread .thread"
     And I should see "I loved this" within "ol.thread"
 
+  Scenario: A leaves a comment, B replies to it, A deletes their comment, B edits the comment, A should not receive a comment edit notification email
+
+    Given the work "Generic Work" by "creator"
+      And a comment "A's comment (to be deleted)" by "User_A" on the work "Generic Work"
+      And a reply "B's comment (to be edited)" by "User_B" on the work "Generic Work"
+      And 1 email should be delivered to "User_A"
+    When I am logged in as "User_A"
+      And I view the work "Generic Work" with comments
+      And I delete the comment
+    When I am logged in as "User_B"
+      And I view the work "Generic Work" with comments
+      And I follow "Edit"
+      And I fill in "Comment" with "B's improved comment (edited)"
+      And I press "Update"
+    Then 1 email should be delivered to "User_A"
```

Comment:
Isn't the goal not to send User_A an email here?

Reply:
Ah, yes, but that 1 email is the "You have a reply!" email from line 125. It checks that there isn't a second email sent to A.
I should add a comment in the test to make this clear.

Reply:
Ah, okay! 

In that case, the best thing to do would be add `And all emails have been delivered` after line 125 to basically reset the count, and then the last step can be `Then 0 emails should be delivered to "User_A"`

Reply:
Oh that's neat - I've always wondered what that does before now!